crates/gosub_renderer/src/render_tree.rs

```diff
@@ -10,40 +10,31 @@ use gosub_shared::byte_stream::{ByteStream, Encoding};
 use std::fs;
 use url::Url;
 
+/// Generates a render tree from the given URL.. if the source is given, the URL is not loaded, but the source HTML is used instead
 pub(crate) async fn load_html_rendertree<
     C: HasRenderTree<LayoutTree = RenderTree<C>, RenderTree = RenderTree<C>> + HasHtmlParser,
 >(
     url: Url,
+    source: Option<&str>,
 ) -> gosub_shared::types::Result<(RenderTree<C>, Fetcher)> {
     let fetcher = Fetcher::new(url.clone());
 
-    let rt = load_html_rendertree_fetcher::<C>(url, &fetcher).await?;
+    let rt = match source {
+        Some(source) => load_html_rendertree_source::<C>(url, source).await?,
+        None => load_html_rendertree_fetcher::<C>(url, &fetcher).await?,
+    };
 
     Ok((rt, fetcher))
 }
```

Comment:
I think this should be removed, instead callers should create their fetcher by temselves and give it `load_html_rendertree_fetcher`

Reply:
Added a new issue for this:

https://github.com/gosub-io/gosub-engine/issues/732